README.md

```diff
@@ -1,51 +1,99 @@
+
+* To use Gazebo simulator, "1.1 PC Setup" is required.
+* To control Mini Pupper, "1.2 Mini Pupper Setup" is required.
+* To control Mini Pupper using visualize tools, "1.1 PC Setup" and "1.2 Mini Pupper Setup" is required.
+
+### 1.1 PC Setup
+
+PC Setup corresponds to PC (your desktop or laptop PC) for controlling Mini Pupper remotely or execute simulator.  
+__Do not apply these PC Setup commands to your Raspberry Pi on Mini Pupper.__
+
+Ubuntu 22.04 + ROS 2 Humble is required.  
+Please follow the [installation document for ROS Humble](https://docs.ros.org/en/humble/Installation/Ubuntu-Install-Debians.html) or use the [unofficial ROS 2 installation script](https://github.com/Tiryoh/ros2_setup_scripts_ubuntu).
+
+After ROS 2 installation, download the Mini Pupper ROS package in the workspace.
 
 ```sh
-cd ~
-mkdir -p ros2_ws/src
-cd ros2_ws/src
+mkdir -p ~/ros2_ws/src
+cd ~/ros2_ws/src
 git clone https://github.com/mangdangroboticsclub/mini_pupper_ros.git -b ros2
-git clone --recursive https://github.com/chvmp/champ -b ros2
-cd ..
+vcs import < minipupper_ros/.minipupper.repos --recursive
+```
+
+Build and install all ROS packages.
+
+```sh
+cd ~/ros2_ws
 rosdep install --from-paths src --ignore-src -r -y
+sudo apt-get install ros-humble-teleop-twist-keyboard
 colcon build --symlink-install
+```
+
+### 1.2 Mini Pupper Setup
+
+Mini Pupper Setup corresponds to the Raspberry Pi on your Mini Pupper.
+
+Ubuntu 22.04 + ROS 2 Humble is required.  
+Please follow the [installation document for ROS Humble](https://docs.ros.org/en/humble/Installation/Ubuntu-Install-Debians.html) or use the [unofficial ROS 2 installation script](https://github.com/Tiryoh/ros2_setup_scripts_ubuntu).
+
+After ROS 2 installation, download the Mini Pupper ROS package in the workspace.
+
+```sh
+mkdir -p ~/ros2_ws/src
+cd ~/ros2_ws/src
+git clone https://github.com/mangdangroboticsclub/mini_pupper_ros.git -b ros2
+vcs import < minipupper_ros/.minipupper.repos --recursive
+# compiling gazebo and cartographer on Raspberry Pi is not recommended
+touch champ/champ/champ_description/AMENT_IGNORE
+touch champ/champ/champ_gazebo/AMENT_IGNORE
+touch champ/champ/champ_navigation/AMENT_IGNORE
+touch mini_pupper_ros/mini_pupper_gazebo/AMENT_IGNORE
+touch mini_pupper_ros/mini_pupper_navigation/AMENT_IGNORE
+```
+
+Build and install all ROS packages.
+
+If the Raspberry Pi has less than 4GB memory, try `MAKEFLAGS=-j1 colcon build --executor sequential --symlink-install` instead of `colcon build --symlink-install`
+
+```sh
+# install dependencies without unused heavy packages
+cd ~/ros2_ws
+rosdep install --from-paths src --ignore-src -r -y --skip-keys=joint_state_publisher_gui --skip-keys=rviz2
 sudo apt-get install ros-humble-teleop-twist-keyboard
+colcon build --symlink-install
 ```
 
-## 2. Test in RViz
+## 2. Quick Start
+
+### 2.1 Test in RViz
```

Comment:
What about adding a note that the steps related to RViz and Gazebo are for PC only?

Reply:
Thanks, added. [ac192f5](https://github.com/mangdangroboticsclub/mini_pupper_ros/pull/40/commits/ac192f577059512f5ed89231aacca29b919f54f2), [c4f3319](https://github.com/mangdangroboticsclub/mini_pupper_ros/pull/40/commits/c4f331951e2305d469d2e0c6ab40efd288b919d8)